include/libtorrent/torrent_info.hpp

```diff
@@ -504,8 +497,8 @@ namespace libtorrent
 		// ``metadata()`` returns a the raw info section of the torrent file. The size
 		// of the metadata is returned by ``metadata_size()``.
 		int metadata_size() const { return m_info_section_size; }
-		boost::shared_array<char> metadata() const
-		{ return m_info_section; }
```

Comment:
It's not obvious that this is correct. previously shared ownership of this buffer was returned, but with this patch only a (non-owning) reference is returned